File: Solutions/0383-ransom-note/solution.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
    map<char, int> cnt;
    for(int i = 0 ; i < magazine.size() ; i++) cnt[magazine[i]]++;
    for(int i = 0 ; i < ransomNote.size() ; i++) {
        char c = ransomNote[i];
        cnt[c]--;
        if(cnt[c] < 0) return false;
    }
    return true;
}
};

File: Solutions/0383-ransom-note/solution.ts
const a_char = "a".charCodeAt(0);

function canConstruct(ransomNote: string, magazine: string): boolean {
    const ran_cnt: number[] = Array(26).fill(0);
    const mg_cnt: number[] = Array(26).fill(0);
    for(let char of ransomNote) {
        ran_cnt[char.charCodeAt(0) - a_char]++;
    }
    for(let char of magazine) {
        mg_cnt[char.charCodeAt(0) - a_char]++;
    }
    for(let i = 0 ; i < 26 ; i++) if(ran_cnt[i] > mg_cnt[i]) return false;
    return true;
};

